frontend/src/app/modules/common/dynamic-forms/services/dynamic-fields/dynamic-fields.service.ts

```diff
@@ -0,0 +1,263 @@
+import { Injectable } from '@angular/core';
+import {
+  IAttributeGroup,
+  IDynamicInputConfig,
+  IFieldSchema,
+  IFieldSchemaWithKey,
+  IOPFormlyFieldConfig,
+  IOPFormModel,
+  IOPFormSchema,
+} from "../../typings";
+import { FormlyFieldConfig } from "@ngx-formly/core";
+import { of } from "rxjs";
+import { map } from "rxjs/operators";
+import { HttpClient } from "@angular/common/http";
+
+
+@Injectable()
+export class DynamicFieldsService {
+  readonly inputsCatalogue:IDynamicInputConfig[] = [
+    {
+      config: {
+        type: 'textInput',
+        className: 'inline-edit--field',
+        templateOptions: {
+          type: 'text',
+        },
+      },
+      useForFields: ['String']
+    },
+    {
+      config: {
+        type: 'integerInput',
+        className: `inline-edit--field`,
+        templateOptions: {
+          type: 'number',
+          locale: I18n.locale,
+        },
+      },
+      useForFields: ['Integer']
+    },
+    {
+      config: {
+        type: 'booleanInput',
+        className: `inline-edit--field inline-edit--boolean-field`,
+        templateOptions: {
+          type: 'checkbox',
+        },
+      },
+      useForFields: ['Boolean']
+    },
+    {
+      config: {
+        type: 'dateInput',
+        className: `inline-edit--field`,
+      },
+      useForFields: ['Date', 'DateTime']
+    },
+    {
+      config: {
+        type: 'formattableInput',
+        className: `textarea-wrapper`,
+        templateOptions: {
+          editorType: 'full',
+          inlineLabel: true,
+        },
+      },
+      useForFields: ['Formattable']
+    },
+    {
+      config: {
+        type: 'selectInput',
+        className: `inline-edit--field`,
+        templateOptions: {
+          type: 'number',
+          locale: I18n.locale,
+          bindLabel: 'title',
+          searchable: false,
+          virtualScroll: true,
+          typeahead: false,
+          clearOnBackspace: false,
+          clearSearchOnAdd: false,
+          hideSelected: false,
+          text: {
+            add_new_action: I18n.t('js.label_create'),
+          },
+        },
+        expressionProperties: {
+          'templateOptions.clearable': (model:any, formState:any, field:FormlyFieldConfig) => !field.templateOptions?.required,
+          'templateOptions.multiple': (model:any, formState:any, field:FormlyFieldConfig) => (field.key! as string).replace('_links.', '').startsWith('[]'),
+        },
+      },
+      useForFields: ['Priority', 'Status', 'Type', 'User', 'Version', 'TimeEntriesActivity', 'Category', 'CustomOption', 'Project', 'ProjectStatus']
+    },
+  ]
+
+  constructor(
+    private _httpClient:HttpClient,
+  ) { }
+
+  getConfig(formSchema:IOPFormSchema, formPayload:IOPFormModel):IOPFormlyFieldConfig[] {
```

Comment:
This should probably be called something like `getConfigForSchema` or something

Reply:
I thought about it but, since the params are quite clear, it seemed enough to me. In order to getConfig for Dynamicfield you need to pass the formSchema and formPayload. I'm not sure about describing the params in the method name, in this case, the result would be ``getConfigForSchemaAndPayload`` and it would change whenever we add/remove any param. What do you think?